### PR TITLE
Add removeAt method

### PR DESCRIPTION
**Pull request by kborucinski**

### CODE REVIEW

File: array/removeAt.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default (index: number) => ([...xs]: any[]) => {
```
Comment (przemyslawzalewski) [inline array/removeAt.ts]: Please spread within the body to keep the interface as `(index: number) => (xs: any[]): any[] => { ... }

File: array/removeAt.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import removeAt from "./removeAt.ts.js";
+
+describe("removeAt", () => {
+  it("should remove the element specified by index", () => {
+    expect(removeAt(3)([1, 2, 3, 4, 5, 6])).toEqual([1, 2, 3, 5, 6]);
+  });
+
+  it("should return empty array when given empty array", () => {
```
Comment (przemyslawzalewski) [inline array/removeAt.test.js]: Please add a test case for out-of-range index value on non-empty array as well.

File: array/removeAt.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import removeAt from "./removeAt.ts.js";
```
Comment (przemyslawzalewski) [inline array/removeAt.test.js]: The import is broken and makes the checks to fail:

```suggestion
import removeAt from "./removeAt.ts";
```

File: array/removeAt.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import removeAt from "./removeAt.ts";
```
Comment (przemyslawzalewski) [inline array/removeAt.test.js]: Tests must be written in TS - please rename the file to `removeAt.test.ts`.

File: array/removeAt.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "removeAt",
+  "description": "Removes the element specified by index.",
+  "signature": "(index: number) => (xs: any[]) => any[]",
+  "examples": [
+    {
+      "language": "javascript",
+      "content": "removeAt(3)([1, 2, 3, 4, 5, 6]) // ⇒ [1, 2, 3, 5, 6]"
+    }
+  ],
+  "questions": ["TODO: List questions that may this function answer."]
```
Comment (przemyslawzalewski) [inline array/removeAt.json]: Consider adding this: "How to remove an item at a particular index?"

File: array/removeAt.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "removeAt",
+  "description": "Removes the element specified by index.",
```
Comment (przemyslawzalewski) [inline array/removeAt.json]: Consider rephrasing this into:

```suggestion
  "description": "Removes an element at the given index from the given array.",
```

File: array/removeAt.test.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+describe("removeAt", () => {
+  it("should remove the element specified by index", () => {
```
Comment (przemyslawzalewski) [inline array/removeAt.test.js]: Consider this:

```suggestion
  it("should remove an element at the specified index", () => {
```

File: array/removeAt.test.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    expect(removeAt(3)([1, 2, 3, 4, 5, 6])).toEqual([1, 2, 3, 5, 6]);
+  });
+
+  it("should return source array when given index is out of range", () => {
```
Comment (przemyslawzalewski) [inline array/removeAt.test.js]: Consider simplifying this into:

```suggestion
  it("should do nothing when given an index out of range", () => {
```
